source/math/FloatConvexHull2D.ooc

```diff
@@ -57,18 +61,77 @@ FloatConvexHull2D: class {
 		this contains(box leftBottom) && this contains(box rightBottom) && this contains(box leftTop) && this contains(box rightTop)
 	}
 	transform: func (transform: FloatTransform2D) -> This {
-		// TODO: Re-compute convex hull from new points if necessary, i.e. if transform changes order of points
 		newPoints := VectorList<FloatPoint2D> new(this count)
 		for (i in 0 .. this count)
 			newPoints add(transform * this _points[i])
 		This new(newPoints)
 	}
-	_isOnLeft: static func (leftPoint, rightPoint, queryPoint: FloatPoint2D) -> Bool {
-		(rightPoint x - leftPoint x) * (queryPoint y - leftPoint y) > (rightPoint y - leftPoint y) * (queryPoint x - leftPoint x)
+	computeHull: func {
+		if (this count > 2) {
+			// Uses the Quickhull algorithm if more than two points, average complexity O(n log n)
+			// https://en.wikipedia.org/wiki/Quickhull
+			points := this _points copy()
+			this _points clear()
+			leftMostIndex := 0
+			rightMostIndex := 0
+			for (i in 0 .. points count)
+				if (points[i] x < points[leftMostIndex] x)
+					leftMostIndex = i
+				else if (points[i] x > points[rightMostIndex] x)
+					rightMostIndex = i
+			leftEndpoint := points[leftMostIndex]
+			rightEndpoint := points[rightMostIndex]
+			leftSet := VectorList<FloatPoint2D> new()
+			rightSet := VectorList<FloatPoint2D> new()
+			for (i in 0 .. points count)
+				if (i != leftMostIndex && i != rightMostIndex)
+					if (This _isOnLeft(points[leftMostIndex], points[rightMostIndex], points[i]))
```

Comment:
Why not use `leftEndpoint` and `rightEndpoint` instead of again accessing them from the list?